### PR TITLE
shaderObject: Fix creating shaders with incompatible binary

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Make sure the correct VkResult is returned

The issue was that VkResult could be overwritten if it was anything other than VK_SUCCESS

### CODE REVIEW

File: layers/shader_object.cpp
Comment (jeremyg-lunarg) [inline layers/shader_object.cpp:3100]: this variable should be snake_case

**Review by jeremyg-lunarg — CHANGES_REQUESTED** [review]

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM, I'll let you merge this when you're sure it is ready.

**Comment by jeremyg-lunarg** [comment]
please rebase to pick up CI fixes

**Review by daniel-story — APPROVED** [review]

**Pull request merged**
